File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/transport/rest/dto/ChatCompletionChoiceDto.java
package com.sisqueslabs.nexora.api.contexts.chat.transport.rest.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public record ChatCompletionChoiceDto(
        int index,
        ChatCompletionMessageDto message,
        @JsonProperty("finish_reason") String finishReason) {
}

File: src/contexts/chat/transport/rest/dtos/chat-completion-choice.dto.ts
import { ApiProperty } from '@nestjs/swagger';

import { ChatCompletionMessageDto } from './chat-completion-message.dto';

export class ChatCompletionChoiceDto {
  @ApiProperty({ example: 0 })
  index!: number;

  @ApiProperty({ type: ChatCompletionMessageDto })
  message!: ChatCompletionMessageDto;

  @ApiProperty({ example: 'stop' })
  finish_reason!: string;
}
